### PR TITLE
feat: support Cloud SQL's new field `server_ca_mode`.

### PR DESCRIPTION
**Pull request by feng-zhe**
This PR supports the new field `settings.ip_configuration.server_ca_mode` of Cloud SQL instances.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
sql: added `server_ca_mode` field to `google_sql_database_instance` resource
```

### CODE REVIEW

**Comment by feng-zhe** [comment]
@trodge Hi Thomas, do you know why the above  `presubmit-generate-diffs (graphite-docker-images)` failed? I see it complains about `go.mod missing autogen` but I don't know what causes it. I updated the `go.mod` to use the newer `google.golang.org/api` version 190. Thanks.

File: mmv1/third_party/terraform/go.mod.erb [hunk 1]
```diff
@@ -1,10 +1,9 @@
-<% autogen_exception -%>
```
Comment (trodge) [inline mmv1/third_party/terraform/go.mod.erb:1]: This tag is needed to prevent the ruby compiler from treating this as a normal template.

Comment (feng-zhe) [reply mmv1/third_party/terraform/go.mod.erb]: Thanks. Just fixed it.

**Comment by feng-zhe** [comment]
@trodge Hi Thomas, I believe the above failed ACC tests are not related to this PR. I see some of them failing in other PRs too. Thanks.

File: mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown [hunk 2]
```diff
@@ -384,6 +384,8 @@ This setting can be updated, but it cannot be removed after it is set.
     * For MySQL instances, use the same value pairs as the PostgreSQL instances.
     * For SQL Server instances, set it to `ALLOW_UNENCRYPTED_AND_ENCRYPTED` when `require_ssl=false` and `ENCRYPTED_ONLY` otherwise.
 
+* `server_ca_mode` - (Optional) Specify how the server certificate's Certificate Authority is hosted. Supported values are `GOOGLE_MANAGED_INTERNAL_CA`, and `GOOGLE_MANAGED_CAS_CA`.
```
Comment (trodge) [inline mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:387]:
```suggestion
* `server_ca_mode` - (Optional) Specify how the server certificate's Certificate Authority is hosted. Supported value is `GOOGLE_MANAGED_INTERNAL_CA`.
```

**Review by trodge — CHANGES_REQUESTED** [review]
Just one small nit, otherwise LGTM.

**Review by trodge — APPROVED** [review]

**Comment by shuyama1** [comment]
@feng-zhe @trodge we've had to revert this feature due to a consistent test failure during our nightly integration tests. The issue seems to be related to updated dependencies in this PR. I'll investigate further today to determine the next steps.

**Pull request merged**